src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -242,9 +242,7 @@ public final synchronized void shutdown() throws InterruptedException
          LOGGER.info("{} - is closing down.", poolName);
          logPoolState("Before closing ");
 
-         connectionBag.close();
-         softEvictConnections();
-         addConnectionExecutor.shutdownNow();
+         addConnectionExecutor.shutdown();
```

Comment:
The bag is closed first to prevent connections from being added.  When the ``addConnection()`` calls ``concurrentBag.add()`` an ``IllegalStateException`` will be thrown, and that exception will be caught and the connection forced closed.